Add `toArray` method generation

PrintNow opened the pull request:
Preview URL (Deployed on Cloudflare Pages): https://feat-add-toarray-method.json2php-generator.pages.dev

---

- Add PhpClassToArrayMethodPresenter for generating `toArray` method
- Remove unnecessary hasReadonlyProperties check
- Clean up setter generation logic

The toArray method allows converting PHP objects back to arrays, which is a commonly requested feature for data transfer.

Refs #48

PrintNow committed:
feat: add toArray method generation

- Add PhpClassToArrayMethodPresenter for generating toArray method

The toArray method allows converting PHP objects back to arrays,
which is a commonly requested feature for data transfer.

Refs #48

PrintNow committed:
test: add test cases for toArray method

- Add test case for basic toArray method functionality
- Add test case for combined fromJson and toArray methods

The test cases cover all PHP versions (7.3-8.3) and verify the correct
generation of toArray method in PHP classes.

jacobdekeizer reviewed (changes requested):
Thank you for your contribution, I've added some feedback.

PrintNow committed:
refactor: remove unnecessary returnType variable in PhpClassToArrayMethodPresenter

- Directly use 'array' string in openMethod call
- Improve code conciseness

Refs: https://github.com/jacobdekeizer/json-to-php-generator/pull/49/commits/51d97d704f67258f1b8515003af18839da74f46d#r2152841737

PrintNow committed:
fix: conditionally add docblock in PhpClassToArrayMethodPresenter

- Only add @return docblock when propertyDocblock setting is 'all'
- Follow the same docblock logic as other methods

Refs: https://github.com/jacobdekeizer/json-to-php-generator/pull/49/commits/51d97d704f67258f1b8515003af18839da74f46d#r2152840135

PrintNow committed:
test: update test cases for toArray method

PrintNow commented on src/presenters/PhpClassToArrayMethodPresenter.ts:
```diff
@@ -0,0 +1,103 @@
+            '@return array'
+        ];
+        
+        codeWriter.writeMultilineDocblock(docblockLines);
```
> The docblock should only be added if the propertyDocblock setting is set to all: https://github.com/jacobdekeizer/json-to-php-generator/blob/master/src/dto/Settings.ts#L88

Fixed! 👍

jacobdekeizer committed:
Use presenter for correct property access

jacobdekeizer commented:
@PrintNow I've made an additional change so the correct letter casing for the property name is used. Thank you!

PrintNow commented:
@jacobdekeizer Oops, missed that casing — thanks for catching and correcting it! Really appreciate it. 😊

atva-dev commented:
Thanks @PrintNow @jacobdekeizer
